### ISSUE TITLE
Failed to open... to expose in sandbox errors

### ISSUE THREAD

**Issue by philn** [sonnyp/Tangram#186]
As mentioned in https://github.com/sonnyp/Tangram/issues/151 creating a new tab is not working as expected, the web page does not load.

**Workaround**

`WEBKIT_FORCE_SANDBOX=0 flatpak run re.sonny.Tangram`

**Tangram version**

1.4.0 Powered by
WebKitGTK 2.36.0
gjs 1.72.0
libsoup 2.74.2

**Linux distribution**

`Fedora 35 Silverblue`

**Flatpak version**

1.12.7

**Desktop environment**

`GNOME Shell 41.5 Wayland`

**Comment by sonnyp** [comment]
I don't think it's an issue with Tangram itself but we may be able to help debug

Can it be reproduced with Epiphany flatpak?

Can it be reproduced with the latest Flatpak?
https://copr.fedorainfracloud.org/coprs/gasinvein/flatpak-edge/packages/ 

Can anyone reproduce this on something else than Fedora Silverblue

**Comment by tjapro** [comment]
There is some more lines with my report @sonnyp :

Also my system is fully update, right before the Fedora 36 release (the Fedora 36 Beta release was March 28, 2022 = https://fedoramagazine.org/announcing-fedora-36-beta/):

```
OS	- Fedora Linux 35 (Workstation Edition) x86_64 
DE	- GNOME 41.4 
WM	- Mutter  
Kernel	- 5.16.19-200.fc35.x86_64
```

The flatpak program info:

```
Tangram - Browser for your pinned tabs

          ID: re.sonny.Tangram
         Ref: app/re.sonny.Tangram/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 1.4.0
     License: GPL-3.0+
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 158,7 kB
     Runtime: org.gnome.Platform/x86_64/42
         Sdk: org.gnome.Sdk/x86_64/42

      Commit: 35b1a223b0a36ec47155d878fe0c1ba650ce2d9e3ec351531171e63bc2368b1b
      Parent: d8b20c341e20588a33c86d154d918c48d28c88691fe76297615dda0dd1b5b648
     Subject: Update GNOME runtime to 42 (#18) (5f547083)
        Date: 2022-03-24 19:04:39 +0000
```

The flatpak logs:
```
flatpak run re.sonny.Tangram

Gjs-Message: 18:10:59.833: JS LOG: gjs 1.72.0
Gjs-Message: 18:10:59.833: JS LOG: WebKitGTK 2.36.0
Gjs-Message: 18:10:59.833: JS LOG: libsoup 2.74.2
Gjs-Message: 18:10:59.834: JS LOG: env: flatpak
Gjs-Message: 18:10:59.834: JS LOG: data_dir: /home/.../.var/app/re.sonny.Tangram/data/Tangram
Gjs-Message: 18:10:59.834: JS LOG: cache_dir: /home/.../.var/app/re.sonny.Tangram/cache/Tangram
Gjs-Message: 18:10:59.834: JS LOG: config_dir: /home/.../.var/app/re.sonny.Tangram/config/Tangram
Gjs-Message: 18:10:59.834: JS LOG: applications_dir: /home/.../.local/share/applications
Gjs-Message: 18:10:59.834: JS LOG: keyfile_settings_path: 
Gjs-Message: 18:10:59.845: JS LOG: programInvocationName: /app/bin/re.sonny.Tangram
Gjs-Message: 18:10:59.845: JS LOG: _: null
Gjs-Message: 18:10:59.847: JS LOG: argv /app/bin/re.sonny.Tangram
Gtk-Message: 18:10:59.886: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:10:59.886: Failed to load module "pk-gtk-module"
Gtk-Message: 18:10:59.888: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:10:59.888: Failed to load module "pk-gtk-module"
Gtk-Message: 18:11:00.909: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:11:00.909: Failed to load module "pk-gtk-module"
Gtk-Message: 18:11:00.911: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:11:00.912: Failed to load module "pk-gtk-module"
Gtk-Message: 18:11:00.923: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:11:00.923: Failed to load module "pk-gtk-module"
Gtk-Message: 18:11:00.925: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:11:00.925: Failed to load module "pk-gtk-module"
Gjs-Message: 18:11:05.128: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 18:11:05.129: JS LOG: navigation,OTHER,https://web.whatsapp.com/
Gjs-Message: 18:11:05.129: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 18:11:05.129: JS LOG: navigation,OTHER,https://web.whatsapp.com/
Gtk-Message: 18:11:05.522: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:11:05.522: Failed to load module "pk-gtk-module"
Gtk-Message: 18:11:05.524: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:11:05.524: Failed to load module "pk-gtk-module"
Gjs-Message: 18:11:05.886: JS LOG: decide-policy,RESPONSE
Gjs-Message: 18:11:05.888: JS LOG: decide-policy,RESPONSE

--> Here I tried to open twice instagram.com

Failed to open /home/.../.var/app/re.sonny.Tangram/data/Tangram/7331bfc1113b430f99b0a017fb75ef70 to expose in sandbox: Ficheiro ou pasta inexistente

(process:11989): libsoup-WARNING **: 18:11:20.360: Can't open /home/.../.var/app/re.sonny.Tangram/data/Tangram/7331bfc1113b430f99b0a017fb75ef70/cookies.sqlite
Failed to open /home/.../.var/app/re.sonny.Tangram/data/Tangram/7331bfc1113b430f99b0a017fb75ef70 to expose in sandbox: Ficheiro ou pasta inexistente

--> Here I tried to open twice facebook.com

Failed to open /home/.../.var/app/re.sonny.Tangram/data/Tangram/7331bfc1113b430f99b0a017fb75ef70 to expose in sandbox: Ficheiro ou pasta inexistente
Failed to open /home/.../.var/app/re.sonny.Tangram/data/Tangram/7331bfc1113b430f99b0a017fb75ef70 to expose in sandbox: Ficheiro ou pasta inexistente

--> Clicked in the Cancel button

(re.sonny.Tangram:2): Gjs-CRITICAL **: 18:11:44.734: JS ERROR: Gio.IOErrorEnum: Failed to open /home/.../.var/app/re.sonny.Tangram/data/Tangram/7331bfc1113b430f99b0a017fb75ef70

--> Program closed

destroy@file:///app/share/Tangram/instances.js:90:40
onCancelNewTab@file:///app/share/Tangram/window.js:220:20
main@file:///app/share/Tangram/main.js:59:22
@file:///app/bin/re.sonny.Tangram:18:5
```

I do not tried the workaround, because it will disable sandboxing and 

> browser tabs will no longer be subsandboxed separately from each other

which is a not good for me: it will break the separation of whatsapp accounts.

Also, as this comment says https://github.com/flathub/org.gnome.Epiphany/issues/21#issuecomment-814079653 :
> Since there are no changes to make in Epiphany, I'm going to close this in favor of any of [flatpak/flatpak#3514](https://github.com/flatpak/flatpak/issues/3514) [flatpak/flatpak#3563](https://github.com/flatpak/flatpak/issues/3563) [flatpak/flatpak#3712](https://github.com/flatpak/flatpak/issues/3712), which are all duplicates of each other. Once those are fixed, Epiphany will stop crashing. In the meantime, workaround is to manually disable the sandbox or not use mixed locales.

And those issues are closed, and one of them are just for tracking.

_Originally posted by @tjapro in https://github.com/sonnyp/Tangram/issues/151#issuecomment-1085239459_

**Comment by tjapro** [comment]
> Can it be reproduced with Epiphany flatpak?

Yes. No problem.

Flatpak info:
```
Web - Navegador Web para o GNOME

          ID: org.gnome.Epiphany
         Ref: app/org.gnome.Epiphany/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 42.0
     License: GPL-3.0+
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 60,6 MB
     Runtime: org.gnome.Platform/x86_64/42
         Sdk: org.gnome.Sdk/x86_64/42

      Commit: fb828f5faa8bff4a0daabd5c1df4927ddd0965d9e43e73e21093359e4c56f46b
      Parent: faf25e62815908888e1fb16b819571b0a2f9f3518b995ccc8644b8a06c528c96
     Subject: Update to 42.0 (63bc8a12)
        Date: 2022-03-22 19:47:34 +0000
```

Flatpak logs:
```
flatpak run org.gnome.Epiphany

Gtk-Message: 18:18:39.492: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:18:39.492: Failed to load module "pk-gtk-module"
Gtk-Message: 18:18:39.493: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:18:39.493: Failed to load module "pk-gtk-module"
Gtk-Message: 18:18:41.253: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:18:41.254: Failed to load module "pk-gtk-module"
Gtk-Message: 18:18:41.257: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:18:41.258: Failed to load module "pk-gtk-module"

--> Opened instagram.com

Gtk-Message: 18:18:56.281: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:18:56.282: Failed to load module "pk-gtk-module"
Gtk-Message: 18:18:56.283: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:18:56.283: Failed to load module "pk-gtk-module"

--> Accept cookies

Gtk-Message: 18:19:08.644: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:19:08.644: Failed to load module "pk-gtk-module"
Gtk-Message: 18:19:08.645: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:19:08.645: Failed to load module "pk-gtk-module"

--> Changed to facebook.com

Gtk-Message: 18:19:12.505: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:19:12.506: Failed to load module "pk-gtk-module"
Gtk-Message: 18:19:12.507: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:19:12.507: Failed to load module "pk-gtk-module"

--> Program closed

Gtk-Message: 18:19:24.565: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:19:24.565: Failed to load module "pk-gtk-module"
Gtk-Message: 18:19:24.566: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:19:24.567: Failed to load module "pk-gtk-module"
```

**Comment by tjapro** [comment]
WIP

> Can it be reproduced with the latest Flatpak?

Yes, I'm preparing the system ...

**Comment by sonnyp** [comment]
> which is a not good for me: it will break the separation of whatsapp accounts.

No - it doesn't break "containerization" but it does disable a security measure.

**Comment by tjapro** [comment]
> https://github.com/sonnyp/Tangram/issues/186#issuecomment-1100246271

```
WEBKIT_FORCE_SANDBOX=0 flatpak run re.sonny.Tangram

Gjs-Message: 18:34:58.721: JS LOG: gjs 1.72.0
Gjs-Message: 18:34:58.721: JS LOG: WebKitGTK 2.36.0
Gjs-Message: 18:34:58.721: JS LOG: libsoup 2.74.2
Gjs-Message: 18:34:58.721: JS LOG: env: flatpak
Gjs-Message: 18:34:58.722: JS LOG: data_dir: /home/.../.var/app/re.sonny.Tangram/data/Tangram
Gjs-Message: 18:34:58.722: JS LOG: cache_dir: /home/.../.var/app/re.sonny.Tangram/cache/Tangram
Gjs-Message: 18:34:58.722: JS LOG: config_dir: /home/.../.var/app/re.sonny.Tangram/config/Tangram
Gjs-Message: 18:34:58.722: JS LOG: applications_dir: /home/.../.local/share/applications
Gjs-Message: 18:34:58.722: JS LOG: keyfile_settings_path: 
Gjs-Message: 18:34:58.735: JS LOG: programInvocationName: /app/bin/re.sonny.Tangram
Gjs-Message: 18:34:58.735: JS LOG: _: null
Gjs-Message: 18:34:58.736: JS LOG: argv /app/bin/re.sonny.Tangram
Gtk-Message: 18:34:58.780: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:34:58.780: Failed to load module "pk-gtk-module"
Gtk-Message: 18:34:58.783: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:34:58.783: Failed to load module "pk-gtk-module"
Gtk-Message: 18:34:59.078: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:34:59.078: Failed to load module "pk-gtk-module"
Gtk-Message: 18:34:59.080: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:34:59.081: Failed to load module "pk-gtk-module"
Gtk-Message: 18:34:59.145: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:34:59.145: Failed to load module "pk-gtk-module"
Gtk-Message: 18:34:59.147: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:34:59.147: Failed to load module "pk-gtk-module"
Gjs-Message: 18:34:59.445: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 18:34:59.446: JS LOG: navigation,OTHER,https://web.whatsapp.com/
Gjs-Message: 18:34:59.523: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 18:34:59.523: JS LOG: navigation,OTHER,https://web.whatsapp.com/
Gtk-Message: 18:34:59.681: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:34:59.681: Failed to load module "pk-gtk-module"
Gtk-Message: 18:34:59.683: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:34:59.683: Failed to load module "pk-gtk-module"
Gjs-Message: 18:35:00.240: JS LOG: decide-policy,RESPONSE
Gjs-Message: 18:35:00.244: JS LOG: decide-policy,RESPONSE

--> Opened the new tab window.

(process:13576): libsoup-WARNING **: 18:35:28.555: Can't open /home/.../.var/app/re.sonny.Tangram/data/Tangram/297145babb93482b862e8222f5151585/cookies.sqlite
Gtk-Message: 18:35:28.595: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:35:28.596: Failed to load module "pk-gtk-module"
Gtk-Message: 18:35:28.600: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:35:28.601: Failed to load module "pk-gtk-module"

--> Searched for instagram.com

Gjs-Message: 18:35:28.815: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 18:35:28.815: JS LOG: navigation,OTHER,tangram-resource:///re/sonny/Tangram/data/blank.html
Gjs-Message: 18:35:28.857: JS LOG: decide-policy,RESPONSE
Gjs-Message: 18:35:36.480: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 18:35:36.481: JS LOG: navigation,OTHER,http://instagram.com/
Gjs-Message: 18:35:37.178: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 18:35:37.178: JS LOG: navigation,OTHER,https://instagram.com/
Gjs-Message: 18:35:37.794: JS LOG: decide-policy,NAVIGATION_ACTION
Gjs-Message: 18:35:37.794: JS LOG: navigation,OTHER,https://www.instagram.com/

** (process:67349): WARNING **: 18:35:39.162: Error writing credentials to socket: Erro ao enviar a mensagem: Túnel quebrado
Gtk-Message: 18:35:39.192: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:35:39.193: Failed to load module "pk-gtk-module"
Gtk-Message: 18:35:39.195: Failed to load module "canberra-gtk-module"
Gtk-Message: 18:35:39.195: Failed to load module "pk-gtk-module"
Gjs-Message: 18:35:39.416: JS LOG: decide-policy,RESPONSE

--> Clicked in the Cancel button

(re.sonny.Tangram:2): Gjs-CRITICAL **: 18:35:46.738: JS ERROR: Gio.IOErrorEnum: Trash portal failed on /home/.../.var/app/re.sonny.Tangram/data/Tangram/297145babb93482b862e8222f5151585
destroy@file:///app/share/Tangram/instances.js:90:40
onCancelNewTab@file:///app/share/Tangram/window.js:220:20
main@file:///app/share/Tangram/main.js:59:22
@file:///app/bin/re.sonny.Tangram:18:5
```

**Comment by sonnyp** [comment]
Ok I can reproduce.

I will investigate, for now, yopu can use this as a workaround:

```
flatpak override --user --filesystem=home re.sonny.Tangram
```

**Comment by sonnyp** [comment]
Fixed in v1.4.1

Will be available on Flathub shortly.
